packages/@ember/-internals/metal/lib/property_get.ts

```diff
@@ -134,9 +135,9 @@ export function get(obj: object, keyName: string): any {
   return value;
 }
 
-export function _getPath<T extends object>(root: T, path: string): any {
+export function _getPath<T extends object>(root: T, path: string | string[]): any {
   let obj: any = root;
-  let parts = path.split('.');
+  let parts = typeof path === 'string' ? path.split('.') : path;
```

Comment:
I think it will be even more performant if `_getPath` only accept array of path, https://github.com/emberjs/ember.js/pull/17166/files#diff-45589baa0b0954cefba717cf63c270f9R100 and here you can put `isObjectLike ? _getPath(obj, keyName.split('.')) : undefined `